jwst/lib/set_telescope_pointing.py

```diff
@@ -405,88 +627,94 @@ def update_wcs_from_telem(
+        model.meta.guidestar.gs_v3_pa_science
+    )
+    t_pars.pcs_mode = model.meta.guidestar.gs_pcs_mode
+    logger.debug('guide_star_wcs from model: %s', t_pars.guide_star_wcs)
+    logger.debug('PCS_MODE: %s', t_pars.pcs_mode)
+
+    # Get jwst velocity
+    t_pars.jwst_velocity = np.array([
+        model.meta.ephemeris.velocity_x_bary,
+        model.meta.ephemeris.velocity_y_bary,
+        model.meta.ephemeris.velocity_z_bary,
+    ])
+    logger.debug('JWST Velocity: %s', t_pars.jwst_velocity)
+
+    # Retrieve tracking mode.
+
     # Get the pointing information
     try:
-        pointing = get_pointing(obsstart, obsend, engdb_url=engdb_url,
-                                tolerance=tolerance, reduce_func=reduce_func)
+        pointing = get_pointing(obsstart, obsend, engdb_url=t_pars.engdb_url,
+                                tolerance=t_pars.tolerance, reduce_func=t_pars.reduce_func)
     except ValueError as exception:
-        if not allow_default:
+        if not t_pars.allow_default:
             raise
         else:
             logger.warning(
                 'Cannot retrieve valid telescope pointing.'
                 ' Default pointing parameters will be used.'
-                '\nException is {}'.format(exception)
             )
+            logger.warning('Exception is %s', exception)
             logger.info("Setting ENGQLPTG keyword to PLANNED")
             model.meta.visit.engdb_pointing_quality = "PLANNED"
     else:
         # compute relevant WCS information
         logger.info('Successful read of engineering quaternions:')
-        logger.info('\tPointing = {}'.format(pointing))
-        model.meta.visit.engdb_pointing_quality = "CALCULATED"
+        logger.info('\tPointing: %s', pointing)
         try:
-            wcsinfo, vinfo = calc_wcs(pointing, siaf, **transform_kwargs)
-            logger.info("Setting ENGQLPTG keyword to CALCULATED")
+            t_pars.pointing = pointing
+            wcsinfo, vinfo, transforms = calc_wcs(t_pars)
+            pointing_engdb_quality = f'CALCULATED_{t_pars.method.value.upper()}'
+            logger.info('Setting ENGQLPTG keyword to %s', pointing_engdb_quality)
+            model.meta.visit.pointing_engdb_quality = pointing_engdb_quality
         except Exception as e:
             logger.warning(
                 'WCS calculation has failed and will be skipped.'
                 'Default pointing parameters will be used.'
-                '\nException is {}'.format(e)
             )
-            if not allow_default:
+            logger.warning('Exception is %s', e)
+            if not t_pars.allow_default:
                 raise
             else:
                 logger.info("Setting ENGQLPTG keyword to PLANNED")
-                model.meta.visit.engdb_pointing_quality = "PLANNED"
-    logger.info('Aperture WCS info: {}'.format(wcsinfo))
-    logger.info('V1 WCS info: {}'.format(vinfo))
+                model.meta.visit.pointing_engdb_quality = "PLANNED"
```

Comment:
I hope/assume that the model attribute name has also been updated in the datamodels schema?

Reply:
Yes. Should be the code right below these statements.